imports/api/questions.js

```diff
@@ -195,84 +197,56 @@ if (Meteor.isServer) {
     } else this.ready()
   })
 
-  // question library for a user (student can see those they own or created)
-  Meteor.publish('questions.library', function () {
+  Meteor.publish('questions.library', function (courseId) {
     if (this.userId) {
       const user = Meteor.users.findOne({_id: this.userId})
-      if (user.hasRole(ROLES.admin)) {
-        return Questions.find({
-          approved: true,
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false} })
-      } else if (user.isInstructorAnyCourse()) {
-        const courses = _.pluck(Courses.find({instructors: this.userId}).fetch(), '_id')
-        return Questions.find({
-          '$or': [{owner: this.userId}, {courseId: { '$in': courses }, approved: true}],
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false} })
-      } else {
-        // students. By checking for creator, they can see the questions they submitted
-        // that have been moved to a course library (which changes the owner w/o copying)
-        return Questions.find({
-          '$or': [{creator: this.userId}, {owner: this.userId}],
-          sessionId: {$exists: false} })
+      const course = Courses.findOne({ _id: courseId})
+      let query = { 
+        sessionId: {$exists: false},
+        courseId: courseId,
       }
-    } else this.ready()
-  })
-
-  Meteor.publish('questions.libraryInCourse', function (courseId) {
-    if (this.userId) {
-      const user = Meteor.users.findOne({_id: this.userId})
       if (user.hasRole(ROLES.admin)) {
-        return Questions.find({
-          approved: true,
-          studentCopyOfPublic: {$exists: false},
-          sessionId: {$exists: false},
-          courseId: courseId
-         })
+        query = _.extend({ studentCopyOfPublic: {$exists: false} }, query)
       } else if (user.isInstructor(courseId)) {
-        return Questions.find({
-          owner: this.userId,
-          courseId: courseId,
-          approved: true,
-          sessionId: {$exists: false}
-        })
+        query = _.extend({ owner: this.userId }, query)
```

Comment:
For an instructor or admin, we want approved:true and studentCopyOfPublic doesn't exist.
For an instructor,  we should not check if owner is this.userId, since we want to be able to see all approved questions in the course.